From a610f4c21b083c0f9cf62ad2251dfadc98abb50e Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 22 Jul 2023 15:36:09 +0000
Subject: [PATCH] alsactl: add define to compile with glibc 2.38

strlcat and strlcpy have been added to glibc 2.38.
update the defines to use the glibc versions, and not conflict with
string.h.

ref:
- https://sourceware.org/git/?p=glibc.git;a=commit;h=454a20c8756c9c1d55419153255fc7692b3d2199

Fixes: https://github.com/alsa-project/alsa-utils/pull/225
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

Upstream-Status: Backport [https://github.com/alsa-project/alsa-utils/commit/d6a71bfbde9e1710743d3a446c6ea3b41c45234e]
---
 alsactl/init_sysdeps.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/alsactl/init_sysdeps.c b/alsactl/init_sysdeps.c
index 3aca1b4..f09b1ae 100644
--- a/alsactl/init_sysdeps.c
+++ b/alsactl/init_sysdeps.c
@@ -18,6 +18,7 @@
  */
 
 #if defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
+#if !(__GLIBC_PREREQ(2, 38))
 static size_t strlcpy(char *dst, const char *src, size_t size)
 {
 	size_t bytes = 0;
@@ -60,4 +61,5 @@ static size_t strlcat(char *dst, const char *src, size_t size)
 	*q = '\0';
 	return bytes;
 }
+#endif /* !(__GLIBC_PREREQ(2, 38)) */
 #endif /* __GLIBC__ */
